client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java

```diff
@@ -137,14 +149,22 @@ public int getSpaceIdFromCache(String name) {
      * @return
      */
     public List<SpaceNameID> listSpaces() {
+
         ListSpacesReq request = new ListSpacesReq();
-        ListSpacesResp response;
+        ListSpacesResp response = null;
         try {
-            response = client.listSpaces(request);
+            while (retry-- >= 0) {
+                response = client.listSpaces(request);
+                freshClient(response.getCode(), response.getLeader());
```

Comment:
here will do retry times whether it succeeds or fails. You need to jump out of the loop when succeeding.

Reply:
right, stupid retry :(